src/features/game/lib/goblinMachine.ts

```diff
@@ -189,7 +191,15 @@ export function startGoblinVillage(authContext: AuthContext) {
                 onChainState.limitedItems
               );
 
-              return { state: game, limitedItems: limitedItemsById, sessionId };
+              // get Treasury items
+              const treasury = await getTreasuryItems();
```

Comment:
Even if a player is not visiting the treasury they would need to wait for this to resolve. This could add an extra 1-2 seconds to every player's loading time.

I would suggest only loading this data when the Storage room is opened

Reply:
call inside `StorageItems.tsx` instead?

Reply:
Called `getTreasuryItems()` inside StorageItems component. Also added loading UI